controllers/agentmachine_controller.go

```diff
@@ -142,18 +153,14 @@ func (r *AgentMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	// If the AgentMachine doesn't have an agent, find one and set the agentRef
-	if agentMachine.Status.AgentRef == nil {
+	if agentName := agentMachine.Annotations[AgentRefNameAnnotation]; agentName == "" {
```

Comment:
Can we implement this in `findAgent` where we look by the agent label and make that higher priority than any other agent if it matches?

Also we'd need to ensure we're clearing the agent label when we're actually done with it.